kernel/src/actions/crc.rs

```diff
@@ -94,14 +93,46 @@ pub(crate) struct DeletedRecordCountsHistogram {
     pub(crate) deleted_record_counts: Vec<i64>,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub(crate) struct ProtocolMetadata {
```

Comment:
Seems like we're just using this here to return a protocol and metadata? We have lots of other places where we pair them together as either arguments, or a `(protocol, metadata)`. I'd say here can we just do:

```rust
fn try_protocol_and_metadata_from_crc(path: FileMeta, engine: &dyn Engine) -> DeltaResult<(Protocol, Metadata)> {}
```

We could pass through everything and unify on a single struct in the future, but for now this is just confusing